shop/src/main/java/ro/msg/learning/shop/service/CategoryService.java

```diff
@@ -0,0 +1,49 @@
+package ro.msg.learning.shop.service;
+
+import ro.msg.learning.shop.repository.exception.CategoryNotFoundException;
+import ro.msg.learning.shop.model.ProductCategory;
+import org.springframework.web.bind.annotation.*;
+import ro.msg.learning.shop.repository.ProductCategoryRepository;
+
+import java.util.List;
+
+public class CategoryService {
+    private final ProductCategoryRepository productCategoryRepository;
+
+    public CategoryService(ProductCategoryRepository productCategoryRepository) {
+        this.productCategoryRepository = productCategoryRepository;
+    }
+
+    @GetMapping("/productCategory")
```

Comment:
Don't use request mappings in the service layer. The service layer is used to contain the business logic and to facilitate the communication between the controller and the persistence layer. We define the request mappings only on the controller level. 
Remove also the @PathVariable and @RequestBody annotations from the parameters of your methods as they are also applicable only to methods of Spring controllers.
This applies also to the other services.